_posts/2021-02-19-26-rln-relay.md

```diff
@@ -0,0 +1,168 @@
+---
+layout: post
+name:  "Integrating rate limiting nullifiers into Waku Relay to enable spam-protection."
+title:  "Integrating rate limiting nullifiers into Waku Relay to enable spam-protection."
+date:   2021-02-25 12:00:00 +0800
+author: staheri14
+published: true
+permalink: /waku-v2-rln-relay
+categories: research
+summary: A simplified explanation of how Waku RLN-Relay mitigates the spamming issue in a p2p manner.
+image: 
+discuss: 
+---
+
+# Introduction
+
+This blog is going to give you an overview of how we enabled spam protection in waku relay through rate-limiting nullifiers or rln for short.
+
+Firstly, let me give a little background about Waku. Waku is a privacy-preserving peer-to-peer (p2p) messaging protocol for resource restricted devices. Being p2p means that Waku relies on **No** central server, instead messages get transmitted in the network by users' (i.e., peers) collaboration. The underlying protocol (as the writeup of this blog) is [GossipSub](https://docs.libp2p.io/concepts/publish-subscribe/). In a very informal way, the message transmission in GossipSub is like I send a message to you, you send it to whoever you know, and it goes on till the message reaches its destination. 
+
+## Spam Definition
+What does spamming stand for? In a messaging system, spam and spammer can have multiple definitions depending on context e.g., spam can be merely an unsolicited message, or spamming may mean sending an unsolicited message (spam) to large numbers of recipients (in which a large number of the recipients is emphasized). However, in a decentralized p2p system (in specific a p2p system with a GossipSub routing structure), the problem of spamming goes beyond just the recipients of a message and it affects all the peers. This is because the spam messages get routed through the collaboration of all the peers thus consume the resources of the intermediate peers; the system gets congested, and the availability of peers degrades as they get busy spending their computational power/bandwidth/storage capacity on processing spam messages.  
+
+Following the preceding argument, we define a spammer as an entity who uses the messaging system (regardless of the intention) to publish a large number of messages (regardless of the content and the number of recipients) in a short amount of time. The messages issued in this way are called spam. 
```

Comment:
Ditto, I think these paragraphs can be cut down to essentials and rewritten for clarity. Distinguish two cases and define what we mean.